src/react-components/input/ToolbarMicButton.js

```diff
@@ -0,0 +1,50 @@
+import React, { useRef, useCallback } from "react";
+import PropTypes from "prop-types";
+import { FormattedMessage } from "react-intl";
+import { ToolbarButton } from "../input/ToolbarButton";
+import { useVolumeMeter } from "../misc/useVolumeMeter";
+
+export function ToolbarMicButton({ scene, ...rest }) {
+  const ref = useRef();
+  const setRef = useCallback(
+    node => {
+      ref.current = node;
+    },
+    [ref]
+  );
+  const update = useCallback(
+    level => {
+      const node = ref.current;
+      if (node) {
+        const rect = node.querySelector("rect");
```

Comment:
Not an issue with this PR, since this was existing code, but we may want to look into the cost of this `querySelector` call at some point. It's unfortunate that we need to query for the rect every update.

Reply:
Addressed int he previous comment using setRef.